src/Monitor/Tracing/Zipkin.hs

```diff
@@ -259,15 +259,25 @@ b3ToHeaderValue (B3 traceID spanID isSampled isDebug mbParentID) =
       (_ , True) -> "d"
       (True, _) -> "1"
       (False, _) -> "0"
-    required = [encodeTraceID traceID, encodeSpanID spanID, state]
+    required = [encodeZipkinTraceID traceID, encodeSpanID spanID, state]
     optional = encodeSpanID <$> maybeToList mbParentID
   in BS.intercalate "-" $ fmap T.encodeUtf8 $ required ++ optional
 
+-- | Takes into account that the first position of the b3 value is the 32 or 16 lower-hex character TraceId.
```

Comment:
Can you start the new doc comments with a quick blurb describing what the function does (before the specifics sentence here)? Something like "Decodes a trace ID from its Zipkin header representation, returning nothing if it is invalid. ...".